sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -740,7 +767,34 @@ func (host *nodeLanguageHost) execNodejs(ctx context.Context, req *pulumirpc.Run
 		opentracing.Tag{Key: "args", Value: nodeargs})
 	defer tracingSpan.Finish()
 
-	if err := cmd.Run(); err != nil {
+	run := func() error {
+		err := cmd.Start()
+		if err != nil {
+			return err
+		}
+		if req.GetAttachDebugger() {
+			debugConfig, err := structpb.NewStruct(map[string]interface{}{
+				"name":             "Pulumi: Program (Node.js)",
+				"type":             "node",
+				"request":          "attach",
+				"processId":        cmd.Process.Pid,
```

Comment:
Is this right? I think for nodejs you attach to a websocket, using an address with a UUID like `ws://127.0.0.1:9229/bb312e33-9329-48db-acc2-5f870d8bcd08`. I would think that's what we need to send in the DAP request.

Reply:
Looking at more docs, for VSCode at least it seems to uses the process.Id to send SIGUSR1 to node, which activates debug mode on a running node process, but in our case we are starting in debug mode already, so it's a bit confusing to me.

Reply:
I wonder if we need something like
```json
{
  "name": "Attach to Process",
  "type": "node",
  "request": "attach",
  "port": 9229
}
```

https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_setting-up-an-attach-configuration

Reply:
Playing around some more with VSCode and attaching to node process, it might just always send the SIGUSR1 and then attach to the port, so maybe the current code is good.

Reply:
Potentially either of those could work I think, but the `processId` definitely works in both vscode and emacs (which is the one non-vscode editor where I tested this).